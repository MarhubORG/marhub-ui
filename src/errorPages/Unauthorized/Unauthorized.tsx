import React from 'react';
import styled from 'styled-components';
import { Link } from '@reach/router';

export default function Unauthorized(): JSX.Element {
  return (
    <StyledDiv>
      Unauthorized user, please click
      <StyledLink to="/login">login</StyledLink> to continue.
    </StyledDiv>
  );
}

const StyledDiv = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 37.5vh;
`;

const StyledLink = styled(Link)`
  margin: 0 0.2rem;
`;
