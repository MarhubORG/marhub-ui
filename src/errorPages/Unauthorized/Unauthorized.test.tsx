import React from 'react';
import renderer from 'react-test-renderer';
import Unauthorized from './Unauthorized';

it('renders correctly', () => {
  const tree = renderer.create(<Unauthorized />).toJSON();
  expect(tree).toMatchSnapshot();
});
