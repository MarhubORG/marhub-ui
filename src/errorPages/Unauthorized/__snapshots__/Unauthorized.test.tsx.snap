// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 0.2rem;
  margin: 2rem 3rem;
  min-width: 90%;
  min-height: 65vh;
  box-shadow: 0 0.5rem 1rem 0 rgba(44,51,73,0.1);
}

.c1 {
  font-size: 2rem;
}

.c2 {
  margin-top: 0.5rem;
  font-size: 1.2rem;
}

.c3 {
  margin: 0 0.2rem;
}

@media (max-width:768px) {
  .c0 {
    margin: 0rem 0rem;
    min-height: 75vh;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Unauthorized user
  </div>
  <div
    className="c2"
  >
    Click back or click the link to go to the
    <a
      aria-current="page"
      className="c3"
      href="/"
      onClick={[Function]}
    >
      home page
    </a>
    .
  </div>
</div>
`;
